packages/twenty-front/src/modules/workflow/components/WorkflowSingleRecordPicker.tsx

```diff
@@ -0,0 +1,242 @@
+import { useRecordPicker } from '@/object-record/relation-picker/hooks/useRecordPicker';
+import { RecordPickerComponentInstanceContext } from '@/object-record/relation-picker/states/contexts/RecordPickerComponentInstanceContext';
+import { RecordForSelect } from '@/object-record/relation-picker/types/RecordForSelect';
+import { ObjectRecord } from '@/object-record/types/ObjectRecord';
+import { InputLabel } from '@/ui/input/components/InputLabel';
+import { Dropdown } from '@/ui/layout/dropdown/components/Dropdown';
+import { useDropdown } from '@/ui/layout/dropdown/hooks/useDropdown';
+import { DropdownScope } from '@/ui/layout/dropdown/scopes/DropdownScope';
+import SearchVariablesDropdown from '@/workflow/search-variables/components/SearchVariablesDropdown';
+import { isStandaloneVariableString } from '@/workflow/utils/isStandaloneVariableString';
+import { css } from '@emotion/react';
+import styled from '@emotion/styled';
+import { useCallback, useState } from 'react';
+import { isValidUuid } from '~/utils/isValidUuid';
+
+const StyledPlaceholder = styled.div`
+  color: ${({ theme }) => theme.font.color.tertiary};
+  font-size: ${({ theme }) => theme.font.size.md};
+  margin: ${({ theme }) => theme.spacing(2)};
+`;
+
+const StyledFormSelectContainer = styled.div`
+  background-color: ${({ theme }) => theme.background.transparent.lighter};
+  border: 1px solid ${({ theme }) => theme.border.color.medium};
+  border-top-left-radius: ${({ theme }) => theme.border.radius.sm};
+  border-bottom-left-radius: ${({ theme }) => theme.border.radius.sm};
+  border-right: none;
+  border-bottom-right-radius: none;
+  border-top-right-radius: none;
+  box-sizing: border-box;
+  display: flex;
+  overflow: 'hidden';
+  width: 100%;
+  justify-content: space-between;
+  align-items: center;
+  padding-right: ${({ theme }) => theme.spacing(1)};
+`;
+
+const StyledSearchVariablesDropdownContainer = styled.div`
+  align-items: center;
+  display: flex;
+  justify-content: center;
+  ${({ theme }) => css`
+    :hover {
+      background-color: ${theme.background.transparent.light};
+    }
+  `}
+  ${({ theme }) => css`
+    background-color: ${theme.background.transparent.lighter};
+    border-top-right-radius: ${theme.border.radius.sm};
+    border-bottom-right-radius: ${theme.border.radius.sm};
+    border: 1px solid ${theme.border.color.medium};
+  `}
+`;
+
+const StyledRecordChip = styled(RecordChip)`
+  margin: ${({ theme }) => theme.spacing(2)};
+`;
+
+type RecordId = string;
+type Variable = string;
+
+export type WorkflowSingleRecordPickerProps = {
+  label?: string;
+  defaultValue: RecordId | Variable;
+  onChange: (value: RecordId | Variable) => void;
+  objectNameSingular: string;
+};
+
+export const WorkflowSingleRecordPicker = ({
+  label,
+  defaultValue,
+  objectNameSingular,
+  onChange,
+}: WorkflowSingleRecordPickerProps) => {
+  const [draftValue, setDraftValue] = useState<
```

Comment:
naming suggestion 
```suggestion
  const [rawValue, setRawValue] = useState<
```

Reply:
I wanted to be consistent with the other FieldInput created by @Devessier. Why do you think this is better?

Reply:
raw seems more appropriate to me regarding draftValue usage. But it is not a mandatory change and consistency between components is more important. You can resolve